packages/backend/src/Controllers/Ballot/castVoteController.ts

```diff
@@ -71,12 +71,19 @@ async function makeBallotEvent(req: IElectionRequest, targetElection: Election,
     }
 
     //some ballot info should be server-authorative
+    // TODO: move to db trigger
     inputBallot.date_submitted = Date.now();
     if (inputBallot.history == null){
         inputBallot.history = [];
     }
     // preserve the ballot id if it's already provided from prior election
-    if(!inputBallot.ballot_id || targetElection.ballot_source != 'prior_election'){
+    let updatableBallot;
+    if (targetElection.settings.ballot_updates) {
+        updatableBallot = await BallotModel.getBallotByVoterID(roll!.voter_id, inputBallot.election_id, req);
```

Comment:
Interesting! I hadn't used the non-null assertion before 👀

Reply:
To be used sparingly lol